lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -106,50 +105,19 @@ public static ColumnIO getArrayElementColumn(ColumnIO columnIO)
 
     public static Map<List<String>, ColumnDescriptor> getDescriptors(MessageType fileSchema, MessageType requestedSchema)
     {
-        Map<List<String>, ColumnDescriptor> descriptorsByPath = new HashMap<>();
         List<PrimitiveColumnIO> columns = getColumns(fileSchema, requestedSchema);
-        for (String[] paths : fileSchema.getPaths()) {
-            List<String> columnPath = Arrays.asList(paths);
-            getDescriptor(columns, columnPath)
-                    .ifPresent(columnDescriptor -> descriptorsByPath.put(columnPath, columnDescriptor));
-        }
-        return descriptorsByPath;
-    }
-
-    private static Optional<ColumnDescriptor> getDescriptor(List<PrimitiveColumnIO> columns, List<String> path)
-    {
-        checkArgument(!path.isEmpty(), "Parquet nested path should have at least one component");
-        int index = getPathIndex(columns, path);
-        if (index == -1) {
-            return Optional.empty();
-        }
-        PrimitiveColumnIO columnIO = columns.get(index);
-        return Optional.of(columnIO.getColumnDescriptor());
-    }
-
-    private static int getPathIndex(List<PrimitiveColumnIO> columns, List<String> path)
-    {
-        int maxLevel = path.size();
-        int index = -1;
-        for (int columnIndex = 0; columnIndex < columns.size(); columnIndex++) {
-            ColumnIO[] fields = columns.get(columnIndex).getPath();
-            if (fields.length <= maxLevel) {
-                continue;
-            }
-            if (fields[maxLevel].getName().equalsIgnoreCase(path.get(maxLevel - 1))) {
-                boolean match = true;
-                for (int level = 0; level < maxLevel - 1; level++) {
-                    if (!fields[level + 1].getName().equalsIgnoreCase(path.get(level))) {
-                        match = false;
-                    }
-                }
-
-                if (match) {
-                    index = columnIndex;
-                }
+        Map<List<String>, ColumnDescriptor> descriptorsByPath = new HashMap<>(columns.size());
+        for (PrimitiveColumnIO columnIO : columns) {
```

Comment:
I still don't entirely understand it.
Above
```
List<PrimitiveColumnIO> columns = getColumns(fileSchema, requestedSchema);
```
is used to get `columns` against `fileSchema`. Yet here we match `path` again against `fileSchema` too.

I think this definitely deserves a comment

Reply:
Yes, I think even this look-up is unnecessary. I've further simplified the code to build the map directly from `getColumns`